### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by r-erema**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,3 @@
 .idea
 .env
-/bin/
-/pkg/
-/src/github.com/
-/src/golang.org/
-/src/gopkg.in/
+vendor/
```
Comment (vetheslav) [inline .gitignore:3]: If you use a vendor directory, you have to commit it and push in your git repository as well. But I suggest you don't use vendors

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,11 @@
...
-    && rm -rf /wapi_build \
-RUN echo "ok"
-
-WORKDIR /wapi
-
-CMD ./build
+WORKDIR /root/
+COPY --from=builder /var/tmp/wapi/bin/wapi .
+COPY ./.env .
+CMD ["./wapi"]
```
Comment (vetheslav) [inline Dockerfile:11]: Looks good, but you need to start your application with env variables initialized. I don't see any initialization

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+deploy: build
```
Comment (vetheslav) [inline Makefile:1]: It will work, but not for all OSes. Add .phony, please

File: go.sum [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+github.com/labstack/gommon v0.3.0/go.mod h1:MULnywXg0yavhxWKc+lOruYdAhDwPK9wf0OL7NoOu+k=
+github.com/magiconair/properties v1.8.0/go.mod h1:PppfXfuXeibc/6YijjN8zIbojt8czPbwD3XqdrwzmxQ=
+github.com/mattn/go-colorable v0.1.1/go.mod h1:FuOcm+DKB9mbwrcAfNl7/TZVBZ6rcnceauSikq3lYCQ=
+github.com/mattn/go-colorable v0.1.2 h1:/bC9yWikZXAL9uJdulbSfyVNIR3n3trXl+v8+1sx8mU=
+github.com/mattn/go-colorable v0.1.2/go.mod h1:U0ppj6V5qS13XJ6of8GYAs25YV2eR4EVcfRqFIhoBtE=
+github.com/mattn/go-isatty v0.0.5/go.mod h1:Iq45c/XA43vh69/j3iqttzPXn0bhXyGjM0Hdxcsrc5s=
+github.com/mattn/go-isatty v0.0.7/go.mod h1:Iq45c/XA43vh69/j3iqttzPXn0bhXyGjM0Hdxcsrc5s=
+github.com/mattn/go-isatty v0.0.8/go.mod h1:Iq45c/XA43vh69/j3iqttzPXn0bhXyGjM0Hdxcsrc5s=
+github.com/mattn/go-isatty v0.0.9 h1:d5US/mDsogSGW37IV293h//ZFaeajb69h+EHFsv2xGg=
+github.com/mattn/go-isatty v0.0.9/go.mod h1:YNRxwqDuOph6SZLI9vUUz6OYw3QyUt7WiY2yME+cCiQ=
```
Comment (vetheslav) [inline go.sum:108]: It looks like you didn't use `go mod tidy`. Read about it and add this command to you hands-on tools

File: main.go [hunk 5]
```diff
@@ -1,23 +1,24 @@
...
 	"github.com/gorilla/handlers"
 	"github.com/gorilla/mux"
-	"log"
-	"net/http"
-	"os"
-	"time"
+	"github.com/r-erema/wapi/config"
+	"github.com/r-erema/wapi/src/RouteHandler"
+	"github.com/r-erema/wapi/src/Service/Auth"
+	"github.com/r-erema/wapi/src/Service/ConnectionsSupervisor"
```
Comment (vetheslav) [inline main.go:19]: You have to have 3 blocks in your imports: standard libraries, repository packages and then other packages.

File: src/RouteHandler/GetActiveConnectionInfoHandler.go [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
 	"encoding/json"
-	"github.com/Rhymen/go-whatsapp"
-	"github.com/gorilla/mux"
 	"log"
 	"net/http"
+
+	"github.com/Rhymen/go-whatsapp"
+	"github.com/gorilla/mux"
+	"github.com/r-erema/wapi/src/Service/ConnectionsSupervisor"
+	Session2 "github.com/r-erema/wapi/src/Session"
```
Comment (vetheslav) [inline src/RouteHandler/GetActiveConnectionInfoHandler.go:11]: Why do you use 'Session' with capital letter? Rename all filenames and directory names to start with small letters.

File: src/RouteHandler/SendImageHandler.go [hunk 7]
```diff
@@ -48,7 +49,9 @@ func (handler *SendImageHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 		log.Printf("%s: %v\n", errorPrefix, err)
 		return
 	}
-	defer response.Body.Close()
+	defer func() {
+		_ = response.Body.Close()
```
Comment (vetheslav) [inline src/RouteHandler/SendImageHandler.go:53]: It's a really good point! But remember to print a possible error

File: .gitignore (hunk 1 above)
Comment (r-erema) [reply .gitignore:3]: Should we commit only pure vendor directory with gitkeep? Without dependcies, right?
Why it's better to don't use vendors,  is it not welcome in go world?

File: config/config.go [hunk 8]
```diff
@@ -38,9 +36,7 @@ type Config struct {
 	ConnectionTimeout int64
 }
 
-func Init(dotEnvFilePath string) (*Config, error) {
-	_ = godotenv.Load(dotEnvFilePath)
-
+func Init() (*Config, error) {
```
Comment (vetheslav) [inline config/config.go:39]: It's custom for golang to name an init function as you have as 'New'

**Pull request merged**